staging/src/k8s.io/apiserver/pkg/storage/cacher/cacher.go

```diff
@@ -1223,7 +1222,8 @@ func forgetWatcher(c *Cacher, w *cacheWatcher, index int, scope namespacedName,
 		// It's possible that the watcher is already not in the structure (e.g. in case of
 		// simultaneous Stop() and terminateAllWatchers(), but it is safe to call stopLocked()
 		// on a watcher multiple times.
-		c.watchers.deleteWatcher(index, scope, triggerValue, triggerSupported, c.stopWatcherLocked)
+		c.watchers.deleteWatcher(index, scope, triggerValue, triggerSupported)
+		c.stopWatcherLocked(w)
```

Comment:
I'm assuming this is a cleanup change - if so, let's move it to a separate PR...

Reply:
opened https://github.com/kubernetes/kubernetes/pull/121615